src/server/pegasus_server_impl.cpp

```diff
@@ -2113,6 +2139,62 @@ void pegasus_server_impl::update_server_rocksdb_statistics()
     }
 }
 
+void pegasus_server_impl::update_server_rocksdb_rate_limiter_statistics(
+    std::chrono::seconds update_rdb_stat_short_interval)
+{
+    // Update _pfc_rdb_write_limiter_rate_bytes
+    if (_s_rate_limiter) {
+        uint64_t current_total_through = _s_rate_limiter->GetTotalBytesThrough();
+        uint64_t through_rate_per_sec =
+            (current_total_through - _rocksdb_limiter_last_total_through) /
+            update_rdb_stat_short_interval.count();
+        _pfc_rdb_write_limiter_rate_bytes->set(through_rate_per_sec);
+        _rocksdb_limiter_last_total_through = current_total_through;
+        dinfo_f("_pfc_rdb_write_limiter_rate_bytes: {} bytes", through_rate_per_sec);
+    } else {
+        dinfo_f("_pfc_rdb_write_limiter_rate_bytes is 0 bytes because it is disabled");
+    }
+}
+
+void pegasus_server_impl::update_replica_rocksdb_compaction_statistics()
+{
+    std::map<std::string, std::string> props;
+    if (!_db->GetMapProperty(_data_cf, "rocksdb.cfstats", &props)) {
+        dwarn_replica("get compaction statistics failed!");
+        return;
+    }
+    // Update _pfc_rdb_compaction_write_rate_megabytes
+    auto compaction_write_rate_iter = props.find("compaction.Sum.WriteMBps");
+    auto compaction_write_rate = compaction_write_rate_iter == props.end()
+                                     ? 0
+                                     : std::stod(compaction_write_rate_iter->second);
```

Comment:
`stod` is not safe though rocksdb may not return a invalid value, better to use `buf2uint64` instead.